### PR TITLE
LCH-5312: Created filter functions and code refactor

### PR DESCRIPTION
**Pull request by ankitsingh0188**

### CODE REVIEW

File: src/ContentHubClient.php [hunk 1]
```diff
@@ -1182,13 +1183,13 @@ protected function getExceptionResponse($method, array $args, \Exception $except
...
    *   Reason.
-   * @param null $request_id
+   * @param string|null $request_id
    *   The request id from the ContentHub service if available.
    *
    * @return \GuzzleHttp\Psr7\Response
    *   Response.
    */
-  protected function getErrorResponse($code, $reason, $request_id = NULL) {
+  protected function getErrorResponse(int $code, string $reason, ?string $request_id = NULL): Response {
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: Use response interface instead as the return type.

File: src/ContentHubClient.php [hunk 2]
```diff
@@ -1359,13 +1360,85 @@ public function deleteSnapshot($name) {
...
+   *   suffixed with duration unit (m, s, ms etc.).
+   * @param int $size
+   *   Amount of entities to return.
+   *
+   * @return mixed
+   *   Response from backend call.
+   *
+   * @throws \Exception
+   */
+  public function startScrollByFilter(string $filter_uuid, string $scroll_time_window, int $size) {
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: So although the json_decode can return objects too, we are actually command it to return an associative array. Please specify the return type accordingly.

File: src/ContentHubClient.php [hunk 3]
```diff
@@ -1359,13 +1360,85 @@ public function deleteSnapshot($name) {
...
+   *   Scroll id.
+   * @param string $scroll_time_window
+   *   How long the scroll cursor will be retained inside memory.
+   *
+   * @return mixed
+   *   Response from backend call.
+   *
+   * @throws \Exception
+   */
+  public function continueScroll(string $scroll_id, string $scroll_time_window) {
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: Same here about return types.

File: src/ContentHubClient.php [hunk 4]
```diff
@@ -1359,13 +1360,85 @@ public function deleteSnapshot($name) {
...
+   *
+   * @param string $scroll_id
+   *   Scroll id.
+   *
+   * @return mixed
+   *   Response from backend call.
+   *
+   * @throws \Exception
+   */
+  public function cancelScroll(string $scroll_id) {
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: And here.

File: src/ContentHubClient.php [hunk 5]
```diff
@@ -1359,13 +1360,85 @@ public function deleteSnapshot($name) {
...
   }
 
+  /**
+   * Initiates Scroll API request chain.
+   *
+   * @param string $filter_uuid
+   *   Filter uuid to execute by.
+   * @param string $scroll_time_window
+   *   How long the scroll cursor will be retained inside memory. Must be
+   *   suffixed with duration unit (m, s, ms etc.).
```
Comment (saxumVermes) [inline src/ContentHubClient.php:1378]: Hope we can get a response about that soon...

File: test/ContentHubClientTest.php [hunk 6]
```diff
@@ -2412,4 +2412,92 @@ public function testRestoreSnapshotIfSucceeds(): void {
...
+   *
+   * @throws \Exception
+   */
+  public function testContinueScrollIfSucceeds(): void {
+    $scroll_id = 'some-scroll-id';
+    $scroll_time_window = '10m';
+
+    $request_parameters = [
+      'scroll_id' => $scroll_id,
+      'scroll' => '10m',
```
Comment (saxumVermes) [inline test/ContentHubClientTest.php]: You can just use `$scroll_time_window` here too, can't you?

File: test/ContentHubClientTest.php [hunk 7]
```diff
@@ -2412,4 +2412,92 @@ public function testRestoreSnapshotIfSucceeds(): void {
...
+    $response = [
+      'uuid' => 'some-uuid',
+      'request_id' => 'some-request-id',
+      'success' => TRUE,
+    ];
+
+    $this->ch_client
+      ->shouldReceive('delete')
+      ->once()
+      ->with("scroll", ['body' => json_encode($request_parameters)])
```
Comment (saxumVermes) [inline test/ContentHubClientTest.php]: Could you please use single quotes?

**Review by saxumVermes — CHANGES_REQUESTED** [review]

File: src/ContentHubClient.php (hunk 5 above)
Comment (ankitsingh0188) [reply src/ContentHubClient.php:1378]: Yeah, awaiting response from service team.

File: src/ContentHubClient.php [hunk 8]
```diff
@@ -3,27 +3,23 @@
...
 use GuzzleHttp\Exception\BadResponseException;
 use GuzzleHttp\Exception\ClientException;
-use GuzzleHttp\Exception\ConnectException;
 use GuzzleHttp\Exception\RequestException;
-use GuzzleHttp\Exception\ServerException;
 use GuzzleHttp\HandlerStack;
 use GuzzleHttp\Psr7\Response;
 use Psr\Log\LogLevel;
+use Symfony\Component\HttpFoundation\Response as HttpResponse;
 use Symfony\Component\HttpFoundation\Response as SymfonyResponse;
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: I don't understand that, why is that we're importing the same object twice with different names?

**Review by saxumVermes — APPROVED** [review]
Left a comment, there's an unnecessary import, please fix that, otherwise it's good to go.

File: src/ContentHubClient.php (hunk 8 above)
Comment (ankitsingh0188) [reply src/ContentHubClient.php]: Done.

**Review by amangrover90 — APPROVED** [review]
Looks good to me!!

**Pull request merged**